great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -11,4 +46,459 @@ class AbstractDataContext(ABC):
+                logger.info(
+                    "data_context_id is defined globally. Applying override to project_config."
+                )
+                self.config.anonymous_usage_statistics.data_context_id = (
+                    global_data_context_id
+                )
+            else:
+                validation_errors.update(data_context_id_errors)
+        # check for global usage_statistics url
+        global_usage_statistics_url = self._get_global_config_value(
+            environment_variable="GE_USAGE_STATISTICS_URL",
+            conf_file_section="anonymous_usage_statistics",
+            conf_file_option="usage_statistics_url",
+        )
+        if global_usage_statistics_url:
+            usage_statistics_url_errors = anonymizedUsageStatisticsSchema.validate(
+                {"usage_statistics_url": global_usage_statistics_url}
+            )
+            if not usage_statistics_url_errors:
+                logger.info(
+                    "usage_statistics_url is defined globally. Applying override to project_config."
+                )
+                self.config.anonymous_usage_statistics.usage_statistics_url = (
+                    global_usage_statistics_url
+                )
+            else:
+                validation_errors.update(usage_statistics_url_errors)
+        if validation_errors:
+            logger.warning(
+                "The following globally-defined config variables failed validation:\n{}\n\n"
+                "Please fix the variables if you would like to apply global values to project_config.".format(
+                    json.dumps(validation_errors, indent=2)
+                )
+            )
+
+    # static methods
+
+    @staticmethod
+    def _check_global_usage_statistics_opt_out() -> bool:
+        if os.environ.get("GE_USAGE_STATS", False):
+            ge_usage_stats = os.environ.get("GE_USAGE_STATS")
+            if ge_usage_stats in AbstractDataContext.FALSEY_STRINGS:
+                return True
+            else:
+                logger.warning(
+                    "GE_USAGE_STATS environment variable must be one of: {}".format(
+                        AbstractDataContext.FALSEY_STRINGS
+                    )
+                )
+        for config_path in AbstractDataContext.GLOBAL_CONFIG_PATHS:
+            config = configparser.ConfigParser()
+            states = config.BOOLEAN_STATES
+            for falsey_string in AbstractDataContext.FALSEY_STRINGS:
+                states[falsey_string] = False
+            states["TRUE"] = True
+            config.BOOLEAN_STATES = states
+            config.read(config_path)
+            try:
+                if config.getboolean("anonymous_usage_statistics", "enabled") is False:
+                    # If stats are disabled, then opt out is true
+                    return True
+            except (ValueError, configparser.Error):
+                pass
+
+    def _init_stores(self, store_configs: Dict[str, dict]) -> None:
```

Comment:
Nitpick - can this move up in to the private method section?

Reply:
done :)